hbase-common/src/test/java/org/apache/hadoop/hbase/HBaseCommonTestingUtility.java

```diff
@@ -69,7 +69,7 @@
     Compression.Algorithm.NONE, Compression.Algorithm.GZ
   };
 
-  protected Configuration conf;
+  protected final Configuration conf;
```

Comment:
Changing this variable to final break binary compatibility, but is needed to make access to this field threadsafe.

I could maintain binary compatibility by making a _new_ final variable called "originalConf" or something like that which is used by our own internals. but that would still cause a behavior change because if someone made a subclass of these classes that assigned to the `conf` variable we would not pay any attention to that change.

I think we should just release note the break, especially given the on-going issue of "should we be labeling these things IA.Public yet?"

Reply:
> I think we should just release note the break, especially given the on-going issue of "should we be labeling these things IA.Public yet?"

Agreed with putting description on the release notes. And feel weird that these are IA Public, should it be changed to IA Private here?

Reply:
> should it be changed to IA Private here?

no, there 's a whole back and forth community discussion that has been on-going in bits and pieces for years. the short version is that we definitely need something downstream facing for running tests of hbase client code, but no one has provided a clean way to separate that out from how we test our own stuff.

Reply:
s/hbase client code/code that uses hbase client

Reply:
Since we are anyways going to release note the breaking change, can we also make this field `private`? Downstream extended classes will have to mandatorily start using super constructor for setting `conf` anyways and now that it is `final`, we can also make it `private`.

Reply:
we make use of this across our internal subclasses of `HBaseCommonTestingUtility` so if we made it private instead of protected we'd need to add an accessor.